aten/src/ATen/native/Resize.cpp

```diff
@@ -0,0 +1,43 @@
+#include "ATen/ATen.h"
+#include "TH/THTensor.hpp"
+
+namespace at { namespace native {
+
+Tensor& resize__cpu(Tensor& self, IntList size) {
+  if (self.sizes() == size) {
+    return self;
+  }
+
+  auto* self_ = self.unsafeGetTensorImpl();
+
+  int totalSize = 1;
+  int nDimension = size.size();
+  if(nDimension != self_->dim()) {
+    self_->resize_dim(nDimension);
+  }
+
+  for (int d = nDimension - 1; d >= 0; d--) {
+    self_->set_size(d, size[d]);
```

Comment:
Oooh, please don't do it this way. Look at the implementation:

```
  virtual void set_size(int64_t dim, int64_t new_size) {
    sizes_.at(dim) = new_size;
    refresh_numel();
    refresh_contiguous();
  }
```

You're doing way more work than you need to. Use `set_sizes_and_strides` instead.